http.go

```diff
@@ -84,18 +85,15 @@ func (c clientRequest) Post(client *Client, request *soap.SoapMessage) (string,
 		return "", fmt.Errorf("unknown error %s", err)
 	}
 
-	body, err := body(resp)
+	// error in case of incorrect exit code
+	if resp.StatusCode != 200 {
+		return "", fmt.Errorf("http unexpected status: %s", resp.Status)
```

Comment:
The problem here is that if you get an operation timeout (which is also an HTTP 500 error), you'll return an error here that the upper layer will not "ignore" anymore (see [`command.go:134`](https://github.com/masterzen/winrm/blob/master/command.go#L134) and you'll break long running commands.

Note that `OperationTimeout` are SOAP errors, so they need to be parsed.
That's the reason your code breaks the `TestOperationTimeoutSupport` test (the test is designed to show that getting a timeout doesn't mean the command is aborted, it still runs on the server until there are more output to display).

So, I think we first have to check if the body is a `soap+xml`, then error as before, *but* we also have to carry over correctly the error message if it is a straight http error (and not a `soap+xml` error).